omnigibson/utils/usd_utils.py

```diff
@@ -305,12 +347,114 @@ def get_impulses(cls, prim_paths_a, prim_paths_b):
                 from @prim_paths_b
         """
         # Compute subset of matrix and return
-        idxs_a = [cls._PATH_TO_ROW_IDX[path] for path in prim_paths_a]
-        idxs_b = [cls._PATH_TO_COL_IDX[path] for path in prim_paths_b]
-        return cls.get_all_impulses()[idxs_a][:, idxs_b]
+        scene_idx = self._PATH_TO_SCENE_IDX[prim_paths_a[0]]
+        idxs_a = [self._PATH_TO_ROW_IDX[scene_idx][path] for path in prim_paths_a]
+        idxs_b = [self._PATH_TO_COL_IDX[scene_idx][path] for path in prim_paths_b]
+        return self.get_all_impulses(scene_idx)[idxs_a][:, idxs_b]
+
+    def get_contact_pairs(self, scene_idx, row_prim_paths=None, column_prim_paths=None):
```

Comment:
In general (applies to all these new implemented methods), needs docstring. Types of kwargs? What is returned?